src/mapping/NearestNeighborGradientMapping.cpp

```diff
@@ -20,6 +20,10 @@ NearestNeighborGradientMapping::NearestNeighborGradientMapping(
     int        dimensions)
     : NearestNeighborBaseMapping(constraint, dimensions, true, "NearestNeighborGradientMapping", "nng")
 {
+  PRECICE_CHECK(!hasConstraint(CONSERVATIVE), "Nearest-neighbor-gradient mapping is not implemented using a \"conservative\" constraint. Please select constraint=\" consistent\" or a different mapping method.");
+
+  PRECICE_CHECK(!hasConstraint(SCALEDCONSISTENT), "The scaled-consistent mapping hasn't been specifically tested with nearest-neighbor-gradient. Please avoid using it or choose another mapping method. ");
```

Comment:
If you use `PRECICE_CHECK` there is no chance for the user to 'avoid using it', as preCICE will anyway crash all the time. Maybe better use `PRECICE_WARN` and just warn the user that it is not tested.